src/frontends/lean/structure_cmd.cpp

```diff
@@ -1016,6 +1016,14 @@ struct structure_cmd_fn {
     }
 
     expr mk_intro_type() {
+        buffer<expr> params;
+        for (auto param : m_params) {
+            auto type = mlocal_type(param);
+            if (is_class_out_param(type))
+                // hide `out_param`
+                param = update_local(param, app_arg(type), local_info(param));
+            params.push_back(param);
+        }
         expr r = Pi(m_params, mk_intro_type_no_params(), m_p);
```

Comment:
I removed this change (tweaking the intro rule type). It didn't work, and even if it did, it would have broken the preceding commit, where we look for `out_param`s in projection.cpp. If we still want to hide `out_param` when constructing structure instances, we can do it in the instance notation in the elaborator instead.